### PR TITLE
Linux file tailing e2e test

### PR DESCRIPTION
**Pull request by DDuongNguyen**
#### What does this PR do?
This PR introduces a new end-to-end (e2e) test to verify the agent's capability to tail log files on a Linux OS. This serves as a step in our effort to shift our quality assurance testing towards more comprehensive e2e testing.

#### Motivation
This PR purpose is to ensure robustness in the agent's log-tailing feature, especially in various edge cases like file permission changes and log rotations in a Linux environment. 
By converting this QA test to an e2e test, we aim to catch potential issues early in the development cycle in an automated way.

#### Test Detail:
https://datadoghq.atlassian.net/jira/software/projects/AML/boards/2295?selectedIssue=AML-935

### CODE REVIEW

File: test/new-e2e/log-agent/file_tailing_test.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+// logsExampleStackDef returns the stack definition required for the log agent test suite.
+func logsExampleStackDef(vmParams []ec2params.Option, agentParams ...agentparams.Option) *e2e.StackDefinition[e2e.FakeIntakeEnv] {
+	config :=
+		`logs:
+  - type: file
+    path: '/var/log/hello-world.log'
+    service: hello
+    source: custom_log
+`
+	return e2e.EnvFactoryStackDef(
```
Comment (ogaca-dd) [inline test/new-e2e/log-agent/file_tailing_test.go:40]: This code can be written as:
```go
e2e.FakeIntakeStackDef(nil, agentparams.WithLogs(),  agentparams.WithIntegration("custom_logs.d", config))
```

File: test/new-e2e/log-agent/file_tailing_test.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+	fakeintake := s.Env().Fakeintake
+
+	// part 1: no logs
+	// This part verifies that no logs are received initially.
+	err := backoff.Retry(
+
+		func() error {
+			s.Env().VM.ExecuteWithError("sudo rm -f /var/log/hello-world.log")
+			s.Env().VM.ExecuteWithError("sudo rm -f /var/log/hello-world.log.old")
+			s.Env().VM.ExecuteWithError("sudo touch /var/log/hello-world.log")
```
Comment (ogaca-dd) [inline test/new-e2e/log-agent/file_tailing_test.go:93]: Is there a reason for these lines to be inside `backoff.Retry`?

File: test/new-e2e/log-agent/file_tailing_test.go [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+	// This part generates a test log and verifies that it was created successfully.
+
+	var str string
+	for i := 0; i < 100; i++ {
+		str += "hello-world"
+	}
+
+	s.Env().VM.Execute("sudo chmod 777 /var/log/hello-world.log")
+	generateLog := fmt.Sprintf("echo %s > /var/log/hello-world.log", str)
+	_, err = s.Env().VM.ExecuteWithError(generateLog)
```
Comment (ogaca-dd) [inline test/new-e2e/log-agent/file_tailing_test.go:118]: You can use `VM.Execute` so you don't have to check the error.

File: test/new-e2e/log-agent/file_tailing_test.go [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+		service := "hello"
+		logs, err := fakeintake.FilterLogs(service)
+		if err != nil {
+			return err
+		}
+
+		if len(logs) != 1 {
+			m := fmt.Sprintf("no logs with service matching '%s' found, instead got '%s'", service, names)
+			return errors.New(m)
+		}
```
Comment (ogaca-dd) [inline test/new-e2e/log-agent/file_tailing_test.go:145]: Are these lines required as line 148-154 seems to test also this case.

File: test/new-e2e/log-agent/file_tailing_test.go [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+		// check if log's contain matches what is produced
+		logs, err = fakeintake.FilterLogs("hello", fi.WithMessageContaining("hello-world"))
+		if err != nil {
+			return err
+		}
+		if len(logs) != 1 {
+			return fmt.Errorf("received %v logs with 'hello-world', expecting 1", len(logs))
+		}
+
+		// Part 4: Block permission and check the Agent status
```
Comment (ogaca-dd) [inline test/new-e2e/log-agent/file_tailing_test.go:156]: Should Part 4 be in another `backoff.Retry`?

File: test/new-e2e/log-agent/file_tailing_test.go [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+		_, err = s.Env().VM.ExecuteWithError("sudo chmod 000 /var/log/hello-world.log")
```
Comment (ogaca-dd) [inline test/new-e2e/log-agent/file_tailing_test.go:157]: You can use `VM.Execute` instead.
Note: Same comment for the rest of the test

File: test/new-e2e/log-agent/file_tailing_test.go [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+		// Check the Agent status
+		statusOutput, err := s.Env().VM.ExecuteWithError("datadog-agent status | grep -A 10 'custom_logs'")
+		if err != nil || !strings.Contains(statusOutput, "Status: OK") {
+			if strings.Contains(statusOutput, "permission denied") {
+				t.Log("Log file is inaccessible due to permission issues.")
+			} else {
+				t.Log("Log file is inaccessible.")
+			}
+		} else {
+			t.Error("Expected the log file to be inaccessible, but agent status says log status is OK.")
```
Comment (ogaca-dd) [inline test/new-e2e/log-agent/file_tailing_test.go:172]: Can you explain the reason that there is a no assertion but only log?

File: test/new-e2e/log-agent/file_tailing_test.go [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+	)
+}
+
+// TestE2EVMFakeintakeSuite runs the end-to-end test suite for the log agent with virtual machine and fake intake.
+func TestE2EVMFakeintakeSuite(t *testing.T) {
+	e2e.Run(t, &vmFakeintakeSuite{}, logsExampleStackDef(nil), params.WithDevMode())
+}
+
+// TestLogs executes a three-part test of the log agent's interaction with the fake intake system.
+func (s *vmFakeintakeSuite) TestLogs() {
```
Comment (ogaca-dd) [inline test/new-e2e/log-agent/file_tailing_test.go:82]: This test is long. Can you try to split in several tests?

File: test/new-e2e/log-agent/file_tailing_test.go [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+		if len(logs) != 1 {
+			return fmt.Errorf("received %v logs with 'hello-world', expecting 1", len(logs))
+		}
+
+		// Part 4: Block permission and check the Agent status
+		_, err = s.Env().VM.ExecuteWithError("sudo chmod 000 /var/log/hello-world.log")
+		require.NoError(t, err)
+
+		// Sleep 10s to make sure agent status for the log file is updated
+		time.Sleep(10 * time.Second)
```
Comment (ogaca-dd) [inline test/new-e2e/log-agent/file_tailing_test.go:161]: `backoff.Retry` is designed to be used to avoid having to add `time.Sleep`. Maybe this part should in a separate `backoff.Retry`?

**Review by ogaca-dd — CHANGES_REQUESTED** [review]
